base_model_restrict_update/models/ir_model_access.py

```diff
@@ -17,19 +17,26 @@ def check(self, model, mode="read", raise_exception=True):
         res = super(IrModelAccess, self).check(model, mode, raise_exception)
         if self._uid == 1:
             return True
-        self._cr.execute(
-            "SELECT restrict_update FROM ir_model WHERE model = %s", (model,)
-        )
-        query_res = self._cr.dictfetchall()[0]
-        if (
-            query_res["restrict_update"]
-            and mode != "read"
-            and not self.env.user.unrestrict_model_update
-        ):
-            if raise_exception:
+        if mode != "read" and raise_exception:
+            if self._test_readonly(model) or self._test_restrict_update(model):
```

Comment:
Could we exclude some models from this checking? ref. https://gist.github.com/moylop260/43748e848dfc9916fd2f0fd51a69a12a#file-read_only_user-py-L11-L18

Thank you very much for the nice addition.

Reply:
https://github.com/OCA/server-tools/pull/2169#discussion_r745432119